bindings/nodejs/code/stronghold_storage.ts

```diff
@@ -0,0 +1,93 @@
+import { NapiStronghold, NapiDID, NapiKeyLocation, NapiChainState, NapiIdentityState } from '../index.js'
+import { DID, KeyLocation, DIDLease, Signature, ChainState, IdentityState, Storage } from "../../wasm/node/identity_wasm.js";
+
+export class Stronghold implements Storage {
+    private napiStronghold: NapiStronghold;
+
+    constructor(snapshot: string, password: string, dropsave: boolean) {
+        this.napiStronghold = NapiStronghold.create(snapshot, password, dropsave);
+    }
```

Comment:
A constructor can return a promise in JS, although it's considered a bad practice, see [here](https://stackoverflow.com/a/24686979/7090185). I'm not sure why using a `block_on` is bad in this case. But the other option is as you said by using an async static method and throw an error in the constructor, since the class will have the default constructor otherwise.